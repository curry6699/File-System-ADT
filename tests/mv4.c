#include <assert.h>
#include <ctype.h>
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "../Fs.h"

int main(void) {
	Fs fs = FsNew();
	FsMkfile(fs, "hello.txt");
	FsPut(fs, "hello.txt", "hello\n");
	FsMkdir(fs, "world");
	FsMkfile(fs, "world/hello.txt");
	FsPut(fs, "world/hello.txt", "world\n");
	FsTree(fs, NULL);
	printf("---\n");
	FsCat(fs, "hello.txt");
	FsCat(fs, "world/hello.txt");
	printf("---\n");
	char *src[] = { "hello.txt", NULL };
	FsMv(fs, src, "world");
	FsTree(fs, NULL);
	printf("---\n");
	FsCat(fs, "world/hello.txt");

    FsFree(fs);
}