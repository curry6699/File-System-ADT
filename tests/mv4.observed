/
    hello.txt
    world
        hello.txt
---
hello
world
---
/
    hello.txt
    world
        hello.txt
---
world
