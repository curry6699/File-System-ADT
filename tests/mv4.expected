/
    hello.txt
    world
        hello.txt
---
hello
world
---
/
    world
        hello.txt
---
hello
